### ISSUE TITLE
Create Variable Code Action is not accurate for query expression

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-lang#37256]
**Description:**
Consider the following scenario
![Peek 2022-08-04 13-45](https://user-images.githubusercontent.com/35211477/182799088-f2148e88-ac63-44b1-8ee8-26a1edcbff6b.gif)

**Steps to reproduce:**
```Ballerina
public function main() {
    xml x1 = xml `<book>The Lost World</book>`;

    string[] xmlResult = from xml element in x1 select element.toBalString();
}

```
**Affected Versions:**
2201.2.x

**Comment by TharushiJay** [comment]
Blocked by https://github.com/ballerina-platform/ballerina-lang/issues/37450

**Comment by malinthar** [comment]
According to the spec, the compiler can't infer the type of expression `from xml element in x1 select element.toBalString();`. Therefore the expected type is `OTHER`. Therefore the code action should not be suggested. This is fixed with #41117
